GCC gives false-positive array bounds error

On GCC 12.2.0, I get compile errors `-Werror=array-bounds`when dereferencing pointers created from literal constant addresses (like `(type *) 0x0000400`). This causes building the firmware to fail.

It seems that this is because the compiler treats addresses in the first page of memory as an indexed `NULL` pointer, which includes the SDB. This gives the diagnostics like the following:
```
main1.c: In function 'print_cpu_mem_info':
main1.c:120:12: error: array subscript 0 is outside array bounds of 'volatile SystemDataBlock[0]' [-Werror=array-bounds]
  120 |     if (sdb->cpu_model == 0) {
```
where `sdb` is defined as:
```
static volatile SystemDataBlock * const sdb = (volatile SystemDataBlock *)0x400;
```

--- a/code/firmware/rosco_m68k_firmware/Makefile
+++ b/code/firmware/rosco_m68k_firmware/Makefile
@@ -11,7 +11,7 @@ DEFINES:=$(DEFINES)
 CFLAGS=-std=c11 -ffreestanding -nostartfiles -Wall -Wpedantic -Werror	\
        -Iinclude -mcpu=$(CPU) -march=$(ARCH) -mtune=$(TUNE) -Os				\
        -fomit-frame-pointer -fno-delete-null-pointer-checks						\
-			 --param=min-pagesize=0 $(DEFINES)
+       --param=min-pagesize=0 $(DEFINES)
 LDFLAGS=-Map=$(MAP)
 ASFLAGS=-Felf -m$(CPU) -quiet $(DEFINES)
 CC=m68k-elf-gcc

--- a/code/firmware/rosco_m68k_firmware/Makefile
+++ b/code/firmware/rosco_m68k_firmware/Makefile
@@ -11,7 +11,8 @@ DEFINES:=$(DEFINES)
 CFLAGS=-std=c11 -ffreestanding -nostartfiles -Wall -Wpedantic -Werror	\
        -Iinclude -mcpu=$(CPU) -march=$(ARCH) -mtune=$(TUNE) -Os				\
        -fomit-frame-pointer -fno-delete-null-pointer-checks						\
-			 -mno-align-int -mno-strict-align $(DEFINES)
+			 -mno-align-int -mno-strict-align --param=min-pagesize=0				\
+			 $(DEFINES)
 LDFLAGS=-Map=$(MAP)
 ASFLAGS=-Felf -m$(CPU) -quiet $(DEFINES)
 CC=m68k-elf-gcc

--- a/code/firmware/rosco_m68k_firmware/stage2/Makefile
+++ b/code/firmware/rosco_m68k_firmware/stage2/Makefile
@@ -7,10 +7,10 @@ CPU?=68010
 ARCH?=$(CPU)
 TUNE?=$(CPU)
 DEFINES=
-BASE_CFLAGS=-std=c11 -ffreestanding -Wno-unused-parameter     												\
-            -fomit-frame-pointer -Wall -Werror -Wpedantic -Wno-unused-function             \
+BASE_CFLAGS=-std=c11 -ffreestanding -Wno-unused-parameter     															\
+            -fomit-frame-pointer -Wall -Werror -Wpedantic -Wno-unused-function							\
             -Ifat_io_lib -Iinclude -I../include -mcpu=$(CPU) -march=$(ARCH) -mtune=$(TUNE)	\
-            $(DEFINES)
+            --param=min-pagesize=0 $(DEFINES)
 EXTRA_CFLAGS=
 CFLAGS=$(BASE_CFLAGS) -O9
 SIZE_CFLAGS=$(BASE_CFLAGS) -Os

